src/java.base/unix/native/libjava/io_util_md.c

```diff
@@ -162,8 +162,17 @@ fileDescriptorClose(JNIEnv *env, jobject this)
             dup2(devnull, fd);
             close(devnull);
         }
-    } else if (close(fd) == -1) {
-        JNU_ThrowIOExceptionWithLastError(env, "close failed");
+    } else {
+        int result;
+#if defined(_AIX)
+        /* AIX allows close to be restarted after EINTR */
+        RESTARTABLE(close(fd), result);
+#else
+        result = close(fd);
+#endif
+        if (result == -1) {
```

Comment:
This needs to be `if (result == -1 && errno != EINTR)`

Reply:
Yes, I already caught and fixed that gaffe.